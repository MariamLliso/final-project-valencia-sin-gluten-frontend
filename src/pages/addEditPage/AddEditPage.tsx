import { useEffect } from "react";
import { useParams } from "react-router-dom";
import AddEditForm from "../../components/AddEditForm/AddEditForm";
import { useAppDispatch, useAppSelector } from "../../redux/hooks";
import { loadSingleEstablishmentThunk } from "../../redux/thunks/singleEstablishmentThunks/singleEstalbishmentThunks";
import { IUserState } from "../../types/userInterfaces";
import AddEditPageStyle from "./AddEditPageStyle";

const AddEditPage = (): JSX.Element => {
  const { logged }: IUserState = useAppSelector((state) => state.user);
  const dispatch = useAppDispatch();
  const { establishmentId } = useParams();

  useEffect(() => {
    if (logged && establishmentId) {
      dispatch(loadSingleEstablishmentThunk(establishmentId));
    }
  }, [logged, establishmentId, dispatch]);

  return (
    <AddEditPageStyle>
      <AddEditForm />
    </AddEditPageStyle>
  );
};

export default AddEditPage;
