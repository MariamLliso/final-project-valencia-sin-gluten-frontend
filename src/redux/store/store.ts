import { configureStore } from "@reduxjs/toolkit";
import establishmentsSliceReducer from "../features/establishmentsSlice/establishmentsSlice";
import singleEstablishmentSlice from "../features/singleEstablishmentSlice/singleEstablishmentSlice";
import uiSliceReducer from "../features/uiSlice/uiSlice";
import userProfileSliceReducer from "../features/userProfileSlice/userProfileSlice";
import userSliceReducer from "../features/userSlice/userSlice";

const store = configureStore({
  reducer: {
    ui: uiSliceReducer,
    user: userSliceReducer,
    userProfile: userProfileSliceReducer,
    establishments: establishmentsSliceReducer,
    singleEstablishment: singleEstablishmentSlice,
  },
});

export type RootState = ReturnType<typeof store.getState>;
export type AppDispatch = typeof store.dispatch;

export default store;
