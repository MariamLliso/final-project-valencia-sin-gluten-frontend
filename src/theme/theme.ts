import { alpha, createTheme } from "@mui/material";

export const pxToRem = (value: number) => {
  return `${value / 16}rem`;
};

interface IMediaSizes {
  sm: number;
  md: number;
  lg: number;
}

export const responsiveFontSizes = ({ sm, md, lg }: IMediaSizes) => {
  return {
    "@media (min-width:600px)": {
      fontSize: pxToRem(sm),
    },
    "@media (min-width:900px)": {
      fontSize: pxToRem(md),
    },
    "@media (min-width:1200px)": {
      fontSize: pxToRem(lg),
    },
  };
};

export const fontPrimary = "Poppins, sans-serif";
export const fontSecondary = "Libre Bodoni, serif;";

const greyColor = {
  0: "#FFFFFF",
  100: "#F5F5F5",
  200: "#F4F6F8",
  300: "#DFE3E8",
  400: "#C4CDD5",
  500: "#919EAB",
  600: "#637381",
  700: "#454F5B",
  800: "#212B36",
  900: "#161C24",
  500_8: alpha("#919EAB", 0.08),
  500_12: alpha("#919EAB", 0.12),
  500_16: alpha("#919EAB", 0.16),
  500_24: alpha("#919EAB", 0.24),
  500_32: alpha("#919EAB", 0.32),
  500_48: alpha("#919EAB", 0.48),
  500_56: alpha("#919EAB", 0.56),
  500_80: alpha("#919EAB", 0.8),
};

const primaryColor = {
  light: "#E9F3D9",
  main: "#5FAA46",
  dark: "#3b8224",
  contrastText: "#fff",
};

const secondaryColor = {
  lighter: "#D6E4FF",
  light: "#84A9FF",
  main: "#3366FF",
  dark: "#1939B7",
  darker: "#091A7A",
  contrastText: "#fff",
};

const theme = createTheme({
  typography: {
    fontFamily: fontPrimary,
    fontWeightRegular: 400,
    fontWeightMedium: 500,
    fontWeightBold: 600,
    h1: {
      fontFamily: fontSecondary,
      fontWeight: 700,
      lineHeight: 80 / 64,
      fontSize: pxToRem(40),
      ...responsiveFontSizes({ sm: 52, md: 58, lg: 64 }),
    },
    h2: {
      fontFamily: fontSecondary,
      fontWeight: 700,
      lineHeight: 64 / 48,
      fontSize: pxToRem(32),
      ...responsiveFontSizes({ sm: 40, md: 44, lg: 48 }),
    },
    h3: {
      fontFamily: fontSecondary,
      fontWeight: 700,
      lineHeight: 1.5,
      fontSize: pxToRem(24),
      ...responsiveFontSizes({ sm: 26, md: 30, lg: 32 }),
    },
    h4: {
      fontWeight: 700,
      lineHeight: 1.5,
      fontSize: pxToRem(20),
      ...responsiveFontSizes({ sm: 20, md: 24, lg: 24 }),
    },
    h5: {
      fontWeight: 700,
      lineHeight: 1.5,
      fontSize: pxToRem(18),
      ...responsiveFontSizes({ sm: 19, md: 20, lg: 20 }),
    },
    h6: {
      fontWeight: 700,
      lineHeight: 28 / 18,
      fontSize: pxToRem(17),
      ...responsiveFontSizes({ sm: 18, md: 18, lg: 18 }),
    },
    subtitle1: {
      fontWeight: 600,
      lineHeight: 1.5,
      fontSize: pxToRem(16),
    },
    subtitle2: {
      fontWeight: 600,
      lineHeight: 22 / 14,
      fontSize: pxToRem(14),
    },
    body1: {
      lineHeight: 1.5,
      fontSize: pxToRem(16),
    },
    body2: {
      lineHeight: 22 / 14,
      fontSize: pxToRem(14),
    },
    caption: {
      lineHeight: 1.5,
      fontSize: pxToRem(12),
    },
    overline: {
      fontWeight: 700,
      lineHeight: 1.5,
      fontSize: pxToRem(12),
      letterSpacing: 1.1,
      textTransform: "uppercase",
    },
    button: {
      fontWeight: 700,
      lineHeight: 24 / 14,
      fontSize: pxToRem(14),
      textTransform: "uppercase",
    },
  },
  palette: {
    common: { black: "#000", white: "#fff" },
    primary: { ...primaryColor },
    secondary: { ...secondaryColor },
    grey: greyColor,
    text: {
      primary: greyColor[800],
      secondary: greyColor[600],
      disabled: greyColor[500],
    },
    background: { paper: "#fff", default: greyColor[0] },
    action: {
      active: greyColor[600],
      hover: greyColor[500_8],
      selected: greyColor[500_16],
      disabled: greyColor[500_80],
      disabledBackground: greyColor[500_24],
      focus: greyColor[500_24],
      hoverOpacity: 0.08,
      disabledOpacity: 0.48,
    },
  },
  components: {
    MuiButtonBase: {
      styleOverrides: {
        root: {
          letterSpacing: "0.1em",
        },
      },
      defaultProps: {
        disableRipple: true,
      },
    },
    MuiInputBase: {
      styleOverrides: {
        input: {
          "&::placeholder": {
            fontSize: "0.875rem",
          },
        },
      },
    },
    MuiOutlinedInput: {
      styleOverrides: {
        root: {
          borderRadius: `10px`,
          "&.MuiInputBase-multiline": {
            padding: 1,
          },
        },
        input: {
          fontWeight: 500,
          padding: "15.5px 14px",
          borderRadius: `10px`,
          "&.MuiInputBase-inputSizeSmall": {
            padding: "10px 14px",
            "&.MuiInputBase-inputAdornedStart": {
              paddingLeft: 0,
            },
          },
        },
        inputAdornedStart: {
          paddingLeft: 4,
        },
        notchedOutline: {
          borderRadius: `10px`,
        },
      },
    },
  },
});

export default theme;
