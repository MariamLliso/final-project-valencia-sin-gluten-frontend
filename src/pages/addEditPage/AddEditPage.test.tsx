import { Provider } from "react-redux";
import { ThemeProvider } from "styled-components";
import store from "../../redux/store/store";
import theme from "../../theme/theme";
import TestRenderer from "react-test-renderer";
import {
  BrowserRouter,
  MemoryRouter,
  Route,
  Router,
  Routes,
} from "react-router-dom";
import AddEditPage from "./AddEditPage";
import { render, waitFor } from "@testing-library/react";

let mockLogged = true;
const mockDispatch = jest.fn();

jest.mock("react-redux", () => ({
  ...jest.requireActual("react-redux"),
  useSelector: () => ({ logged: mockLogged }),
  useDispatch: () => mockDispatch,
}));

describe("Given AddEditPage component", () => {
  describe("When it's invoked", () => {
    test("Then it should always match this snapshot", () => {
      const testedRegisterPage = TestRenderer.create(
        <BrowserRouter>
          <Provider store={store}>
            <ThemeProvider theme={theme}>
              <AddEditPage />
            </ThemeProvider>
          </Provider>
        </BrowserRouter>
      );

      expect(testedRegisterPage).toMatchSnapshot();
    });
  });

  describe("When it's invoked with logged user and establishmentId", () => {
    test("Then it should call dispatch", async () => {
      const mockPayload = {
        establishmentType: [
          {
            code: "RES",
            description: "Restaurante",
          },
        ],
        name: "Pizza Clik",
        cusine: "Pizzas a domicilio",
        establishmentOffer: [
          {
            code: "DELIVERY",
            description: "A domicilio",
          },
        ],
        adress: "avda aragón 8. bajo 5b",
        municipality: "Valencia",
        region: "Valencia",
        website: "www.pizzaclik.com",
        pictureBackup:
          "https://firebasestorage.googleapis.com/v0/b/valencia-sin-gluten.appspot.com/o/14-6-2022-11-16-IMG_3796.JPG?alt=media&token=90071fab-dc9b-463a-9380-b1799e354de5",
        picture: "images/14-6-2022-11-16-IMG_3796.JPG",
        id: "629c6fab590f5fafee71800b",
      };

      await waitFor(() => {
        const loadSingleEstablishmentAction = {
          type: "singleEstablishment/loadSingleEstablishment",
          payload: mockPayload,
        };

        store.dispatch(loadSingleEstablishmentAction);
      });

      render(
        <MemoryRouter initialEntries={["/establishment/edit/1234"]}>
          <Routes>
            <Route
              path="/establishment/edit/:establishmentId"
              element={
                <Provider store={store}>
                  <ThemeProvider theme={theme}>
                    <AddEditPage />
                  </ThemeProvider>
                </Provider>
              }
            />
          </Routes>
        </MemoryRouter>
      );

      expect(mockDispatch).toHaveBeenCalled();
    });
  });

  describe("When it's invoked with logged user and no establishmentId", () => {
    test("Then it should not call dispatch", async () => {
      const mockPayload = {
        establishmentType: [
          {
            code: "RES",
            description: "Restaurante",
          },
        ],
        name: "Pizza Clik",
        cusine: "Pizzas a domicilio",
        establishmentOffer: [
          {
            code: "DELIVERY",
            description: "A domicilio",
          },
        ],
        adress: "avda aragón 8. bajo 5b",
        municipality: "Valencia",
        region: "Valencia",
        website: "www.pizzaclik.com",
        pictureBackup:
          "https://firebasestorage.googleapis.com/v0/b/valencia-sin-gluten.appspot.com/o/14-6-2022-11-16-IMG_3796.JPG?alt=media&token=90071fab-dc9b-463a-9380-b1799e354de5",
        picture: "images/14-6-2022-11-16-IMG_3796.JPG",
        id: "629c6fab590f5fafee71800b",
      };

      await waitFor(() => {
        const loadSingleEstablishmentAction = {
          type: "singleEstablishment/loadSingleEstablishment",
          payload: mockPayload,
        };

        store.dispatch(loadSingleEstablishmentAction);
      });

      render(
        <MemoryRouter initialEntries={["/establishment/add"]}>
          <Routes>
            <Route
              path="/establishment/add"
              element={
                <Provider store={store}>
                  <ThemeProvider theme={theme}>
                    <AddEditPage />
                  </ThemeProvider>
                </Provider>
              }
            />
          </Routes>
        </MemoryRouter>
      );

      expect(mockDispatch).not.toHaveBeenCalled();
    });
  });
});
