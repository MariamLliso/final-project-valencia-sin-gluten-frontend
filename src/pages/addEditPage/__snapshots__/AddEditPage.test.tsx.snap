// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given AddEditPage component When it's invoked Then it should always match this snapshot 1`] = `
<div
  className="sc-gsnTZi dzQbXI"
>
  <div
    className="sc-bczRLJ dTEFzN"
  >
    <h1
      className="MuiTypography-root MuiTypography-h1 addEdit_tittle css-o2w69a-MuiTypography-root"
    >
      Añadir nuevo establecimiento
    </h1>
    <form
      autoComplete="off"
      onSubmit={[Function]}
    >
      <div
        className="css-1lwbda4-MuiStack-root"
      >
        <h2
          className="MuiTypography-root MuiTypography-h4 addEdit_section-tittle css-5lbw0b-MuiTypography-root"
        >
          Tipo de establecimiento
        </h2>
        <div
          className="MuiFormControl-root css-1nrlq1o-MuiFormControl-root"
        >
          <select
            className="custom-select"
            data-testid="establishmentType-testid"
            id="style"
            name="style"
            onChange={[Function]}
            value=""
          >
            <option
              value=""
            />
            <option
              value="RES"
            >
              Restaurante
            </option>
            <option
              value="CAF"
            >
              Cafetería
            </option>
            <option
              value="OBR"
            >
              Obrador
            </option>
            <option
              value="HOT"
            >
              Hotel
            </option>
            <option
              value="PAS"
            >
              Pastelería
            </option>
            <option
              value="TIE"
            >
              Tienda
            </option>
            <option
              value="HER"
            >
              Herbolarío
            </option>
            <option
              value="SEP"
            >
              Supermercado
            </option>
            <option
              value="CAR"
            >
              Carnicería
            </option>
          </select>
          <p
            className="MuiFormHelperText-root MuiFormHelperText-sizeMedium MuiFormHelperText-contained css-1wc848c-MuiFormHelperText-root"
            id="establishmentType-helpertext"
          >
            <span
              className="notranslate"
            >
              ​
            </span>
          </p>
        </div>
        <h2
          className="MuiTypography-root MuiTypography-h4 addEdit_section-tittle css-5lbw0b-MuiTypography-root"
        >
          Datos del establecimiento
        </h2>
        <div
          className="MuiFormControl-root css-1nrlq1o-MuiFormControl-root"
        >
          <label
            className="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink={false}
            htmlFor="name"
          >
            Nombre
          </label>
          <div
            className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              className="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
              disabled={false}
              id="name"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={false}
              type="text"
              value=""
            />
            <fieldset
              aria-hidden={true}
              className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                className="css-1ftyaf0"
              >
                <span>
                  Nombre
                </span>
              </legend>
            </fieldset>
          </div>
          <p
            className="MuiFormHelperText-root MuiFormHelperText-sizeMedium MuiFormHelperText-contained css-1wc848c-MuiFormHelperText-root"
            id="name-helpertext"
          >
            <span
              className="notranslate"
            >
              ​
            </span>
          </p>
        </div>
        <div
          className="MuiFormControl-root css-1nrlq1o-MuiFormControl-root"
        >
          <label
            className="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink={false}
            htmlFor="cusine"
          >
            Tipo de cocina o productos
          </label>
          <div
            className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              className="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
              disabled={false}
              id="cusine"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={false}
              type="text"
              value=""
            />
            <fieldset
              aria-hidden={true}
              className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                className="css-1ftyaf0"
              >
                <span>
                  Tipo de cocina o productos
                </span>
              </legend>
            </fieldset>
          </div>
          <p
            className="MuiFormHelperText-root MuiFormHelperText-sizeMedium MuiFormHelperText-contained css-1wc848c-MuiFormHelperText-root"
            id="cusine-helpertext"
          >
            <span
              className="notranslate"
            >
              ​
            </span>
          </p>
        </div>
        <label
          className="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth MuiButtonBase-root  css-1fu7jd5-MuiButtonBase-root-MuiButton-root"
          onBlur={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="button"
          tabIndex={0}
        >
          <span
            className="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="AddPhotoAlternateOutlinedIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M18 20H4V6h9V4H4c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2v-9h-2v9zm-7.79-3.17-1.96-2.36L5.5 18h11l-3.54-4.71zM20 4V1h-2v3h-3c.01.01 0 2 0 2h3v2.99c.01.01 2 0 2 0V6h3V4h-3z"
              />
            </svg>
          </span>
          subir una imagen
          <input
            data-testid="upload-image"
            hidden={true}
            id="image"
            onChange={[Function]}
            type="file"
          />
        </label>
        <h2
          className="MuiTypography-root MuiTypography-h4 addEdit_section-tittle css-5lbw0b-MuiTypography-root"
        >
          Dirección
        </h2>
        <div
          className="MuiFormControl-root css-1nrlq1o-MuiFormControl-root"
        >
          <label
            className="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink={false}
            htmlFor="adress"
          >
            Dirección
          </label>
          <div
            className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              className="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
              disabled={false}
              id="adress"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={false}
              type="text"
              value=""
            />
            <fieldset
              aria-hidden={true}
              className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                className="css-1ftyaf0"
              >
                <span>
                  Dirección
                </span>
              </legend>
            </fieldset>
          </div>
          <p
            className="MuiFormHelperText-root MuiFormHelperText-sizeMedium MuiFormHelperText-contained css-1wc848c-MuiFormHelperText-root"
            id="adress-helpertext"
          >
            <span
              className="notranslate"
            >
              ​
            </span>
          </p>
        </div>
        <div
          className="MuiFormControl-root css-1nrlq1o-MuiFormControl-root"
        >
          <label
            className="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink={false}
            htmlFor="municipality"
          >
            Municipio
          </label>
          <div
            className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              className="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
              disabled={false}
              id="municipality"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={false}
              type="text"
              value=""
            />
            <fieldset
              aria-hidden={true}
              className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                className="css-1ftyaf0"
              >
                <span>
                  Municipio
                </span>
              </legend>
            </fieldset>
          </div>
          <p
            className="MuiFormHelperText-root MuiFormHelperText-sizeMedium MuiFormHelperText-contained css-1wc848c-MuiFormHelperText-root"
            id="municipality-helpertext"
          >
            <span
              className="notranslate"
            >
              ​
            </span>
          </p>
        </div>
        <div
          className="MuiFormControl-root css-1nrlq1o-MuiFormControl-root"
        >
          <label
            className="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink={false}
            htmlFor="region"
          >
            Provincia
          </label>
          <div
            className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              className="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
              disabled={false}
              id="region"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={false}
              type="text"
              value=""
            />
            <fieldset
              aria-hidden={true}
              className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                className="css-1ftyaf0"
              >
                <span>
                  Provincia
                </span>
              </legend>
            </fieldset>
          </div>
          <p
            className="MuiFormHelperText-root MuiFormHelperText-sizeMedium MuiFormHelperText-contained css-1wc848c-MuiFormHelperText-root"
            id="region-helpertext"
          >
            <span
              className="notranslate"
            >
              ​
            </span>
          </p>
        </div>
        <h2
          className="MuiTypography-root MuiTypography-h4 addEdit_section-tittle css-5lbw0b-MuiTypography-root"
        >
          Información de contacto
        </h2>
        <div
          className="MuiFormControl-root css-1nrlq1o-MuiFormControl-root"
        >
          <label
            className="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink={false}
            htmlFor="phone"
          >
            Teléfono
          </label>
          <div
            className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
            data-testid="phone-testid"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              className="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
              disabled={false}
              id="phone"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={false}
              type="number"
              value=""
            />
            <fieldset
              aria-hidden={true}
              className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                className="css-1ftyaf0"
              >
                <span>
                  Teléfono
                </span>
              </legend>
            </fieldset>
          </div>
          <p
            className="MuiFormHelperText-root MuiFormHelperText-sizeMedium MuiFormHelperText-contained css-1wc848c-MuiFormHelperText-root"
            id="phone-helpertext"
          >
            <span
              className="notranslate"
            >
              ​
            </span>
          </p>
        </div>
        <div
          className="MuiFormControl-root css-1nrlq1o-MuiFormControl-root"
        >
          <label
            className="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink={false}
            htmlFor="email"
          >
            Correo electrónico
          </label>
          <div
            className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              className="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
              disabled={false}
              id="email"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={false}
              type="text"
              value=""
            />
            <fieldset
              aria-hidden={true}
              className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                className="css-1ftyaf0"
              >
                <span>
                  Correo electrónico
                </span>
              </legend>
            </fieldset>
          </div>
          <p
            className="MuiFormHelperText-root MuiFormHelperText-sizeMedium MuiFormHelperText-contained css-1wc848c-MuiFormHelperText-root"
            id="email-helpertext"
          >
            <span
              className="notranslate"
            >
              ​
            </span>
          </p>
        </div>
        <div
          className="MuiFormControl-root css-1nrlq1o-MuiFormControl-root"
        >
          <label
            className="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink={false}
            htmlFor="website"
          >
            Página web
          </label>
          <div
            className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              className="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
              disabled={false}
              id="website"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={false}
              type="text"
              value=""
            />
            <fieldset
              aria-hidden={true}
              className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                className="css-1ftyaf0"
              >
                <span>
                  Página web
                </span>
              </legend>
            </fieldset>
          </div>
          <p
            className="MuiFormHelperText-root MuiFormHelperText-sizeMedium MuiFormHelperText-contained css-1wc848c-MuiFormHelperText-root"
            id="website-helpertext"
          >
            <span
              className="notranslate"
            >
              ​
            </span>
          </p>
        </div>
        <button
          className="MuiLoadingButton-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButton-fullWidth MuiButtonBase-root css-sppx0e-MuiButtonBase-root-MuiButton-root-MuiLoadingButton-root"
          disabled={false}
          id=":r0:"
          onBlur={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="submit"
        >
          creaer establecimiento
        </button>
      </div>
    </form>
  </div>
</div>
`;
